### DISCUSSION TITLE
Delete action in file / image fields stays the file in filesystem directory non-deleted

### DISCUSSION THREAD

**Discussion by MurzNN** [keystonejs/keystone#6786] [Feature requests]
# Bug report

## Describe the bug

When I attach new file to file field and saves, it successfully appears in files directory. But when I delete the file via Keystone Admin UI, the connection of item with file is deleted, but the file stays in filesystem.

## To Reproduce

1. Launch the `examples-staging/assets-local` example.
2. Create new Post item and attach some image to it.
3. Check the `public/images` directory - file must appear.
4. Edit this Post item and press "Remove" button, Save changes.
5. Check the `public/images` directory - file is still here.
6. Attach this file again to this item, save.
7. Check the directory - you will see two files with same content.

## Expected behaviour

Removed files should be deleted from filesystem too, not only the link in database.

## System information

- Keystone at commit 7a7f3f6d063a58fb7627dbec3be237b4d7e6e26f

**Comment by bladey** [comment]
@MurzNN I'm moving this to our feature request list, as it's not so much a bug, but an enhancement to Keystone.

**Comment by MurzNN** [comment]
Delete action is not deleting the real file from filesystem - is not a bug? :confused: Okay, will wait a fix for this as a feature ;-)

**Comment by MurzNN** [comment]
Any news about this crucial ~bug~ feature?

**Comment by dcousens** [comment]
Hi @MurzNN,
We are actively working on this, this quarter.
We'll ensure to update this discussion as soon as when we have something to share!

**Comment by kenfehling** [comment]
Should the "Change" action also delete the old image? Because otherwise all the old versions of images will accumulate.

**Comment by rachmanov** [comment]
Waiting for this feature, because the use of file fields without it is not possible - a lot of unused files will accumulate on server

**Comment by raveling** [comment]
@MurzNN this is fixed in the latest release https://github.com/keystonejs/keystone/releases/tag/2022-06-09 🙂
